src/terrain/static_mesh.hpp

```diff
@@ -45,10 +46,23 @@ class StaticMesh : public MeshLoader::MultiComplexMesh {
 
     inline StaticMesh(){};
 
+    StaticMesh(entity::Mesh mesh,
+               const std::vector<glm::vec3>& model_transforms);
+
     StaticMesh(
-        std::vector<unsigned short>& indices, std::vector<glm::vec3>& indexed_vertices,
-        std::vector<glm::vec3>& indexed_colors, std::vector<glm::vec3>& indexed_normals,
-        std::vector<glm::vec3>& model_transforms
+        const std::vector<unsigned short>& indices,
+        const std::vector<glm::vec3>& indexed_vertices,
+        const std::vector<glm::vec3>& indexed_colors,
+        const std::vector<glm::vec3>& indexed_normals,
+        const std::vector<glm::vec3>& model_transforms
+    );
+
+    void init(
+        const std::vector<unsigned short>& indices,
+        const std::vector<glm::vec3>& indexed_vertices,
+        const std::vector<glm::vec3>& indexed_colors,
+        const std::vector<glm::vec3>& indexed_normals,
+        const std::vector<glm::vec3>& model_transforms
```

Comment:
Same thing here, why not make `init` into a constructor and fall back to it?

Reply:
fall back to it? I don't think one can call constructors.